server/src/server.ts

```diff
@@ -1,39 +1,218 @@
 import {
 	createConnection,
-	TextDocuments,
-	ProposedFeatures,
+	Diagnostic,
+	DiagnosticSeverity,
+	DidChangeConfigurationNotification,
 	InitializeParams,
-	TextDocumentSyncKind
+	ProposedFeatures,
+	Range,
+	TextDocument,
+	TextDocuments
 } from 'vscode-languageserver';
+import * as ion from 'ion-js';
 
-import {
-	TextDocument
-} from 'vscode-languageserver-textdocument';
-
-// Create a connection for the server, using Node's IPC as a transport.
+// Create a connection for the server. The connection uses Node's IPC as a transport.
 // Also include all preview / proposed LSP features.
 let connection = createConnection(ProposedFeatures.all);
 
-// Create a simple text document manager. 
-let documents: TextDocuments<TextDocument> = new TextDocuments(TextDocument);
+// Create a simple text document manager. The text document manager
+// supports full document sync only
+let documents: TextDocuments = new TextDocuments();
+
+let hasConfigurationCapability: boolean = false;
+let hasWorkspaceFolderCapability: boolean = false;
+let hasDiagnosticRelatedInformationCapability: boolean = false;
 
 connection.onInitialize((params: InitializeParams) => {
+	let capabilities = params.capabilities;
+
+	// Does the client support the `workspace/configuration` request?
+	// If not, we will fall back using global settings
+	hasConfigurationCapability = !!(
+		capabilities.workspace && !!capabilities.workspace.configuration
+	);
+	hasWorkspaceFolderCapability = !!(
+		capabilities.workspace && !!capabilities.workspace.workspaceFolders
+	);
+	hasDiagnosticRelatedInformationCapability = !!(
+		capabilities.textDocument &&
+		capabilities.textDocument.publishDiagnostics &&
+		capabilities.textDocument.publishDiagnostics.relatedInformation
+	);
+
 	return {
 		capabilities: {
-			textDocumentSync: TextDocumentSyncKind.Incremental,
-			// Tell the client that this server supports code completion. 
+			textDocumentSync: documents.syncKind,
+			// Tell the client that the server supports code completion
 			completionProvider: {
-				resolveProvider: false
-			}
+				resolveProvider: true
+			},
+			documentFormattingProvider: true,
+			documentHighlightProvider: true,
 		}
+	};
+});
+
+connection.onInitialized(() => {
+	if (hasConfigurationCapability) {
+		// Register for all configuration changes.
+		connection.client.register(DidChangeConfigurationNotification.type, undefined);
+	}
+	if (hasWorkspaceFolderCapability) {
+		connection.workspace.onDidChangeWorkspaceFolders(_event => {
+			connection.console.log('Workspace folder change event received.');
```

Comment:
Why are we logging this message on change of workspace folder?

Reply:
I will delete this piece of code.